training_record/services.py

```diff
@@ -49,18 +52,20 @@ def create_off_campus_record_from_raw_data(
             contents = []
         if attachments is None:
             attachments = []
+        if role is None or role not in [
```

Comment:
Not `ROUND_CHOICES`. And you don't need to check whether role is None, it could be done in second check, cause `None` won't be in the list.

```python
if role not in (role for role, _ in EventCoefficient.ROLE_CHOICES):
    ...
```